<?xml version="1.0"?>
<launch>
  <arg name="gui" default="true"/>
  <arg name="paused" default="false"/>
  <arg name="world_name" default="$(find nps_uw_sensors_gazebo)/worlds/ocean_waves_watch_whoi_dvl.world"/>

  <!-- use Gazebo's empty_world.launch with uuv_dave_ocean_waves.world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_name)"/>
    <arg name="extra_gazebo_args" value="-s libdsros_sensors.so"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
    <arg name="verbose" value="false"/>
  </include>

  <!-- Load and spawn the model, starts the joint & state publishers, make the model move -->
  <param name="model_name" type="str" value="teledyne_whn"/>
  <param name="base_link_name" type="str" value="dvl_base_link"/>
  <node name="apply_velocity" pkg="nps_uw_sensors_gazebo" type="simple_motion.py" respawn="false" output="screen"/>
</launch>
